src/Apps/Authentication/Utils/helpers.ts

```diff
@@ -159,35 +158,39 @@ export const updateURLWithOnboardingParam = (url: string) => {
   return updatedRedirectTo
 }
 
+const isAbleToTriggerOnboarding = ({
+  type,
+  intent,
+}: {
+  type: ModalType
+  intent?: AuthIntent
+}) => {
+  // Only trigger onboarding for sign ups without a commercial intent
+  return !!(
+    type === ModalType.signup &&
+    intent &&
+    !COMMERCIAL_AUTH_INTENTS.includes(intent)
+  )
+}
+
```

Comment:
cc @damassi — extracted out the logic for triggering onboarding into its own function, which is then also used in the analytics track call. This technically means our analytics data for onboarding is inaccurate. Don't know how much that matters, ultimately.